RandomData.Api/StringGeneration/ServiceImplementations/WordsStringGenerationService.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.Json;
+using RandomData.Api.Services.FileReader;
+using RandomData.Api.Services.Random;
+using RandomData.Api.StringGeneration.Dto;
+using RandomData.Api.StringGeneration.Exceptions;
+using RandomData.Api.StringGeneration.Extensions;
+using RandomData.Api.StringGeneration.Validators;
+
+namespace RandomData.Api.StringGeneration.ServiceImplementations
+{
+    public class WordsStringGenerationService : IStringGenerationService
+    {
+        private readonly IRandom _random;
+        private readonly GetStringParametersValidator _validator = new GetStringParametersValidator();
+        private readonly IEnumerable<string> _words;
+
+        public WordsStringGenerationService(StringGenerationServiceHelpers.StringGenerationServiceOptions options,
+            IFileReader fileReader, IRandom random)
+        {
+            _random = random;
+            var path = options.WordsDictionaryLocation;
+            if (string.IsNullOrEmpty(path)) throw new WordsDictionaryLocationUnspecifiedException();
+            var content = fileReader.GetFileContent(path);
```

Comment:
This service is injected as transient so in each request try to read content of file to memory. This behavior really slow down this endpoint.